### PR TITLE
Reorganize composer scripts

### PR DESCRIPTION
**Pull request by anomiex**
#### Changes proposed in this Pull Request:

This makes several changes to the custom scripts available via composer:

* Most of the phpcs-running commands are renamed to be prefixed with "phpcs:" rather than "php:". The old names are kept as deprecated aliases in all but one case.
* "php:lint" really should lint (`php -l`), not run phpcs.
* Composer always puts vendor/bin/ first in the path, so there's no need to include that in all the script definitions.
* Running phpcs via composer can hit composer's 300-second timeout. We can avoid that by instructing composer not to apply the timeout.
* Commands can build on others, to avoid repeating mostly the same thing (e.g. the timeout disabling) many times.
* We can add documentation for the scripts to the output of `composer list`.

#### Jetpack product discussion

None.

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Try running the various composer commands and see if they still work.

#### Proposed changelog entry for your changes:

* Developer only, none needed.

### CODE REVIEW

**Review by leogermani — APPROVED** [review]
tests and reads well for me! All commands worked as expected

**Comment by anomiex** [comment]
Needs re-approval after the parent was merged.

**Review by jeherve — APPROVED** [review]

**Pull request merged**
